name: CI
run-name: ${{github.actor}} is running this action
on: [pull_request, push]
defaults:
  run:
    working-directory: ./my-app
jobs:
  test-app:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      
      # - name: Install dependencies
      #   run: npm install
      
      - name: Build 
        run: npm  build
      
      # Change the working directory back to the root directory
      - name: Change directory to root
        if: failure()
        run: cd $GITHUB_WORKSPACE

      - name: Create team sync issue
        if: failure()
        run: |
          new_issue_url=$(gh issue create \
            --title "CI Failure in ${{ github.workflow }} - ${{ github.run_id }}" \
            --assignee "$ASSIGNEES" \
            --label "$LABELS" \
            --body "$BODY")
          if [[ $PINNED == true ]]; then
            gh issue pin "$new_issue_url"
          fi
        env:
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          GH_REPO: ${{ github.repository }}
          TITLE: "CI Failure in ${{ github.workflow }} - ${{ github.run_id }}"
          ASSIGNEES: jon65 # Replace with your GitHub username or list of assignees
          LABELS: bug
          BODY: |
            ### CI Failure Details

            An error occurred during the CI workflow.

            - **Job**: ${{ github.job }}
            - **Run Number**: ${{ github.run_number }}
            - **Run ID**: ${{ github.run_id }}
            - **Repository**: ${{ github.repository }}
            - **Branch**: ${{ github.ref }}
            - **Commit**: ${{ github.sha }}
            - **Actor**: ${{ github.actor }}
            - **Workflow**: ${{ github.workflow }}
            - **URL**: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
          PINNED: false
